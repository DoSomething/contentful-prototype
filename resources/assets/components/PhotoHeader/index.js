export PhotoHeader from './PhotoHeader';
export PhotoHeaderContainer from './PhotoHeaderContainer';

export default from './PhotoHeader';
