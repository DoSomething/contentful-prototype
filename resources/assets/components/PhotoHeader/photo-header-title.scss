@import '../../scss/next-toolbox.scss';

.photo-header__title {
  .__secondary {
    position: relative;
    color: $primary-color;
    font-size: $font-large; // TODO: This needs to be the 28px one
    font-weight: $weight-bold;
    text-shadow: $text-shadow;
  }

  .__primary {
    position: relative;
    color: $white;
    letter-spacing: 1.1px;
    font-family: $secondary-font-family;
    font-size: $font-hero;
    font-weight: $weight-normal;
    text-shadow: $text-shadow;

    @include media($medium) {
      font-size: $font-gigantic;
    }
  }
}
