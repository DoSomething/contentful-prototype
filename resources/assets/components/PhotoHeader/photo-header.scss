@import '../../scss/next-toolbox.scss';

.photo-header {
  background-color: $black;
  position: relative;
  text-align: center;
  text-transform: uppercase;
  width: 100%;
}

.photo-header__background-image {
  background-repeat: no-repeat;
  filter: grayscale(1);
  height: 100%;
  left: 0;
  opacity: 0.5;
  position: absolute;
  top: 0;
  width: 100%;
}
