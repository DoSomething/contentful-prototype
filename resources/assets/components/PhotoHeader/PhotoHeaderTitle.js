import React from 'react';
import PropTypes from 'prop-types';
import './photo-header-title.scss';

const PhotoHeaderTitle = ({ primary, secondary }) => (
  <div className="photo-header__title">
    <span className="__secondary">{secondary}</span>
    <h1 className="__primary">{primary}</h1>
  </div>
);

PhotoHeaderTitle.defaultProps = {
  primary: null,
  secondary: null,
};

PhotoHeaderTitle.propTypes = {
  primary: PropTypes.string,
  secondary: PropTypes.string,
};

export default PhotoHeaderTitle;
