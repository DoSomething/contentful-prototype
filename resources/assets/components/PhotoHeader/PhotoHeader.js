import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import './photo-header.scss';

const photoHeaderClasses = 'photo-header background-image-centered padding-lg';

const PhotoHeader = ({ children, className, backgroundImage }) => (
  <div
    className={classnames(photoHeaderClasses, className)}
    style={{ backgroundImage: `url(${backgroundImage})` }}
  >
    { children }
  </div>
);

PhotoHeader.propTypes = {
  backgroundImage: PropTypes.string,
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
};

PhotoHeader.defaultProps = {
  backgroundImage: '',
  className: '',
};

export default PhotoHeader;
