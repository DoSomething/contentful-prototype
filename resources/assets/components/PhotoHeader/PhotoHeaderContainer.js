import { get } from 'lodash';
import { connect } from 'react-redux';

import PhotoHeader from './PhotoHeader';

/**
 * Provide state from the Redux store as props for this component.
 */
const mapStateToProps = state => ({
  themeColor: get(state.campaign.additionalContent, 'themeColor'),
});

/**
 * Export the container component.
 */
export default connect(mapStateToProps)(PhotoHeader);
