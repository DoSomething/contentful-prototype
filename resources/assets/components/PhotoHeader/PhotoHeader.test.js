import React from 'react';
import renderer from 'react-test-renderer';
import PhotoHeader from './PhotoHeader';

test('PhotoHeader snapshot test', () => {
  const tree = renderer.create(
    <PhotoHeader backgroundImage="https://fake-image.com/img.png">
      <h1>title</h1>
    </PhotoHeader>,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
