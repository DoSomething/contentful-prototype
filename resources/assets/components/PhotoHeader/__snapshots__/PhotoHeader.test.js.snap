// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotoHeader snapshot test 1`] = `
<div
  className="photo-header"
  style={
    Object {
      "backgroundImage": "url(https://fake-image.com/img.png)",
    }
  }
>
  <h1>
    title
  </h1>
</div>
`;
